lib/order.rb

```diff
@@ -0,0 +1,92 @@
+require_relative "customer.rb"
+require "csv"
+
+# Build Order Class and instantiate instance variables include readers for all attributes
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+  VALID_STATUSES = [:pending, :paid, :processing, :shipped, :complete]
+
+  # Build initialize method to take in parameters with a default fulfillment status.
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+
+    # Validate that input was among the included possible statuses. Raise error, otherwise.
+    if !VALID_STATUSES.include?(fulfillment_status)
+      raise ArgumentError.new("Invalid Status")
+    end
+    @fulfillment_status = fulfillment_status
+  end
+
+  # total instance method calculates the total cost of any order by summing  products, then adding tax. 
+  def total()
+    total_amount = 0.0
+    @products.each do |product, price|
+      if @products.empty?
+        total_amount = 0.0
+      else total_amount += price.to_f      
+     end
+    end
+    total_amount = (total_amount * 1.075)
+    return (sprintf("%.2f", total_amount).to_f)
```

Comment:
You can use `round` for this:

```suggestion
    return total_amount.round(2)
```